resources/build/tests/trees/build.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -eu
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../build-utils.sh"
+# END STANDARD BUILD SCRIPT INCLUDE
+
+cd "$THIS_SCRIPT_PATH"
+
+# Test builder_describe_outputs and dependencies
+
+builder_describe "parent test module" \
+  :child1 \
+  :child2 \
+  clean \
+  configure \
+  build \
+  test \
+  install \
+  error
+
+builder_parse "$@"
+
+# All child actions will generate files which we need to verify for test
+rm -f ./child?.*
+
+function test_present() {
+  local target=$1
+  local action=$2
+
+  local CHECK="${COLOR_GREEN}✔${COLOR_RESET}" # ✔
+  local CROSS="${COLOR_RED}❌${COLOR_RESET}" # ❌
+
+  if builder_has_action $action:$target; then
+    if [ ! -f $target.$action ]; then
+      builder_die "$CROSS FAIL: ./$target.$action to be present"
+    else
+      echo "$CHECK PASS: ./$target.$action found as expected"
+    fi
+  fi
+}
+
+# We need to specify the order to run actions in the parent script
+#
+# This may looks as simple as:
+#
+#     builder_run_child_actions clean configure build test install
+#
+builder_run_child_actions clean
+test_present child1 clean
+test_present child2 clean
+
+builder_run_child_actions configure
+test_present child1 configure
+test_present child2 configure
+
+# Test chaining commands in a single statement
+builder_run_child_actions build test
+test_present child1 build
+test_present child2 build
+test_present child1 test
+test_present child2 test
+
+# We are customising the order in which child scripts are installed, so 2 goes before 1
+builder_run_child_actions install:child2
+test_present child2 install
+
+builder_run_child_actions install:child1
+test_present child1 install
+
+builder_run_child_actions error
```

Comment:
I'm unclear why this doesn't trigger `builder_start_action error` in both child build.sh scripts.
`trees/build.sh` runs to completion without dieing.

Reply:
`trees/build.sh error` doesn't run to completion; it exits prematurely with exit code 1, which you can test easily enough:

```
mcdurdin@THARK MINGW64 /c/Projects/keyman/app/resources/build/tests/trees (feat/common/builder-parent-child-scripts)
$ ./build.sh error
## [resources/build/tests/trees/child1] error starting...

This error action is supposed to die

## [resources/build/tests/trees/child1] error failed
## [resources/build/tests/trees] error:child1 failed with exit code 1

mcdurdin@THARK MINGW64 /c/Projects/keyman/app/resources/build/tests/trees (feat/common/builder-parent-child-scripts)
$ echo $?
1
```

This means that the second child script doesn't get a chance to run, which is what we're aiming for, because we fail fast.